### PR TITLE
Assigning target

### PR DESCRIPTION
**Pull request by domanttas**
Method and controller for assigning random target along with unit tests for helper and controller.

### CODE REVIEW

File: WhoIsThatServer.Storage/Controllers/TargetElementController.cs [hunk 1]
```diff
@@ -42,5 +42,31 @@ public IHttpActionResult Post([FromBody] TargetElement targetElement)
...
+            }
+
+            catch (ManagerException noPlayersException) when (noPlayersException.ErrorCode == StorageErrorMessages.ThereAreNoPlayersError)
+            {
+                return Json(StorageErrorMessages.ThereAreNoPlayersError);
+            }
+
+            catch (ManagerException targetNotAssignedException) when (targetNotAssignedException.ErrorCode == StorageErrorMessages.TargetNotAssignedError)
+            {
+                return Json(StorageErrorMessages.TargetNotAssignedError);
```
Comment (tomasmuzas) [inline WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Damn, thats looking good! :) one thing. How does the consumer know that there was an error?

Comment (domanttas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Thanks!! Well, I gotta figure it out, but I thought that I could map this error message to frontend and display message there

Comment (tomasmuzas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Well, even though it looks cleaner, you are still pretty much left with the same problem at the frontend/consumer. You need to return different status code (e.g 400), to know that you have failed, if the response in both success and failure cases are strings

Comment (domanttas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: I still have a lot to learn, it seems :D 

But couldn't I just check which string I got from backend in frontend using mapped constants or something?

Comment (tomasmuzas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Not exactly. And strings are especially dangerous. How do you know that I didnt enter "No face detected" as my name? Of course it's a silly example, but when you are asking for name, you should expect all kinds of inputs. So you should definitely know when you have failed by checking status at frontend

Comment (tomasmuzas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: So when you get a not success status, you know you have failed, whatever the error message you got. And THEN map error message to something, if needed

Comment (tomasmuzas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: So your controller should look something like this
```
return Ok(name);
...
return BadRequest(ErrorCode.SomethingHappened);
return BadRequest(ErrorCode.Error2);
```

Comment (domanttas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Fair point, I will look up on how to return bad status codes, as for now I have no idea. But yeah, I understand your point completely, I just need more time to work on it :D

Comment (tomasmuzas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Well, I have provided you an answer already :DD this is the actual way to do that in .NET Core

Comment (domanttas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: Wow, cool. Thanks :D

Comment (domanttas) [reply WhoIsThatServer.Storage/Controllers/TargetElementController.cs:68]: I thought that you provided an abstract example :D

**Pull request merged**